File: rb/features.rb
# Argentofx SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ArgentofxFeatures
  def self.make_feature(name)
    case name
    when "base"
      ArgentofxBaseFeature.new
    when "test"
      ArgentofxTestFeature.new
    else
      ArgentofxBaseFeature.new
    end
  end
end

File: py/argentofx_sdk/features.py
# Argentofx SDK feature factory

from argentofx_sdk.feature.base_feature import ArgentofxBaseFeature
from argentofx_sdk.feature.test_feature import ArgentofxTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ArgentofxBaseFeature(),
        "test": lambda: ArgentofxTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
